cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamFetchHandler.java

```diff
@@ -296,49 +316,65 @@ public P getPosition() {
   }
 
   /**
-   * Creates a {@link ReadFilter} to only read events that are within the given time range.
+   * Returns the events limit for each round of read from the stream reader.
    *
-   * @param startTime Start timestamp for event to be valid (inclusive).
-   * @param endTime   End timestamp fo event to be valid (exclusive).
-   * @return A {@link ReadFilter} with the specific filtering property.
+   * @param count Number of events wanted to read.
+   * @return The actual number of events to read.
    */
-  private ReadFilter createReadFilter(final long startTime, final long endTime) {
-    return new ReadFilter() {
+  private int getReadLimit(int count) {
+    return (count > MAX_EVENTS_PER_READ) ? MAX_EVENTS_PER_READ : count;
+  }
 
-      private long hint;
+  /**
+   * A {@link ReadFilter} for accepting events that are within a given time range.
+   */
+  private static final class TimeRangeReadFilter extends ReadFilter {
+    private final long startTime;
+    private final long endTime;
+    private long hint;
+    private boolean active;
+
+    /**
+     * Creates a {@link TimeRangeReadFilter} with the specific time range.
+     *
+     * @param startTime start timestamp for event to be accepted (inclusive)
+     * @param endTime   end timestamp for event to be accepted (exclusive)
```

Comment:
Remove extra spaces: ``@param endTime end timestamp for event to be accepted (exclusive)``